src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Storage storage;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy, Storage storage) {
+        this.operationStrategy = operationStrategy;
+        this.storage = storage;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.handle(transaction,storage);
```

Comment:
Be careful with direct access to the `Storage` within the `handle` method. Each operation handler should be responsible for manipulating the `Storage` state accordingly. Ensure that the encapsulation principle is followed and that the `Storage` is not exposed or manipulated outside of its intended scope.